src/lib.rs

```diff
@@ -362,8 +368,6 @@ mod explicit {
                         });
                         if is_isolate {
                             valid_isolate_count += 1;
-                        } else {
-                            result.levels[i] = new_level;
```

Comment:
This is an omission in the spec.  Line 543 of the [reference implementation][1] notes this:

```java
// Not really part of the spec
if (!isIsolate) {
    resultLevels[i] = newLevel;
}
```

[The definition of the *directional status stack*][2] implies that the stack's last entry contains the explicit embedding level for the current character:

>  During the pass, the directional status stack always contains entries for all the directional embeddings, overrides, and isolates within which the current position lies – except those that would overflow the depth limit – in addition to the paragraph level entry at the start of the stack. The last entry reflects the innermost valid scope within which the pass's current position lies.

…but the X*n* steps forget to explictly say that the default action is to set the embedding level to the last value on the stack.

[1]: http://www.unicode.org/Public/PROGRAMS/BidiReferenceJava/BidiReference.java
[2]: http://www.unicode.org/reports/tr9/#Explicit_Levels_and_Directions